pkg/ccl/streamingccl/streamproducer/event_stream.go

```diff
@@ -484,6 +500,40 @@ func (s *eventStream) streamLoop(ctx context.Context, frontier *span.Frontier) e
 	}
 }
 
+func (s *eventStream) validateProducerJobAndSpec(ctx context.Context) (roachpb.TenantID, error) {
```

Comment:
@msbutler We might want to add some number of similar checks to the span config stream. We'd need to update its signature to include the main streamID.